src/data/matter.ts

```diff
@@ -1,4 +1,70 @@
+import { UnsubscribeFunc } from "home-assistant-js-websocket";
+import { navigate } from "../common/navigate";
+import { showAlertDialog } from "../dialogs/generic/show-dialog-box";
 import { HomeAssistant } from "../types";
+import { subscribeDeviceRegistry } from "./device_registry";
+
+export const canCommissionMatterExternal = (hass: HomeAssistant) =>
+  hass.auth.external?.config.canCommissionMatter;
+
+export const startExternalCommissioning = (hass: HomeAssistant) =>
+  hass.auth.external?.fireMessage({
+    type: "matter/commission",
+  });
+
+export const addMatterDevice = (element: HTMLElement, hass: HomeAssistant) => {
+  if (!canCommissionMatterExternal(hass)) {
+    showAlertDialog(element, {
+      title: "Use mobile app",
+      text: "Matter commissioning is not supported on this device, use the mobile app to commission Matter devices",
+    });
+    return;
+  }
+
+  let curMatterDevices: Set<string>;
+  let timeout: number | undefined;
+  let unsub: UnsubscribeFunc | undefined = subscribeDeviceRegistry(
```

Comment:
let's move this to it's own function, and we can choose per location if we want this and when to unsubscribe.